polkadot/runtime/parachains/src/assigner_coretime/tests.rs

```diff
@@ -291,6 +291,160 @@ fn assign_core_enforces_higher_or_equal_block_number() {
 	});
 }
 
+#[test]
+fn assign_core_inserts_into_empty_queue() {
+    let core_idx = CoreIndex(0);
+
+    new_test_ext(GenesisConfigBuilder::default().build()).execute_with(|| {
+        run_to_block(1, |_| None);
+
+        // Insert into an empty queue
+        assert_ok!(CoretimeAssigner::assign_core(
+            core_idx,
+            BlockNumberFor::<Test>::from(10u32),
+            vec![(CoreAssignment::Idle, PartsOf57600(28800))],
+            None,
+        ));
+
+        // Verify the assignment
+        assert_eq!(
+            CoreSchedules::<Test>::get((BlockNumberFor::<Test>::from(10u32), core_idx)),
+            Some(Schedule {
+                assignments: vec![(CoreAssignment::Idle, PartsOf57600(28800))],
+                end_hint: None,
+                next_schedule: None
+            })
+        );
+
+        // Verify the queue descriptor
+        assert_eq!(
+            CoreDescriptors::<Test>::get(core_idx).queue,
+            Some(QueueDescriptor {
+                first: BlockNumberFor::<Test>::from(10u32),
+                last: BlockNumberFor::<Test>::from(10u32),
+            })
+        );
+    });
+}
+
+#[test]
+fn assign_core_appends_to_existing_queue() {
+    let core_idx = CoreIndex(0);
+
+    new_test_ext(GenesisConfigBuilder::default().build()).execute_with(|| {
+        run_to_block(1, |_| None);
+
+        // Insert the first assignment
+        assert_ok!(CoretimeAssigner::assign_core(
+            core_idx,
+            BlockNumberFor::<Test>::from(10u32),
+            vec![(CoreAssignment::Idle, PartsOf57600(28800))],
+            None,
+        ));
+
+        // Append a new assignment at a later block number
+        assert_ok!(CoretimeAssigner::assign_core(
+            core_idx,
+            BlockNumberFor::<Test>::from(15u32),
+            vec![(CoreAssignment::Pool, PartsOf57600(28800))],
+            None,
+        ));
+
+        // Verify the new assignment
+        assert_eq!(
+            CoreSchedules::<Test>::get((BlockNumberFor::<Test>::from(15u32), core_idx)),
+            Some(Schedule {
+                assignments: vec![(CoreAssignment::Pool, PartsOf57600(28800))],
+                end_hint: None,
+                next_schedule: None
+            })
+        );
+
+        // Verify the queue descriptor was updated
+        assert_eq!(
+            CoreDescriptors::<Test>::get(core_idx).queue,
+            Some(QueueDescriptor {
+                first: BlockNumberFor::<Test>::from(10u32),
+                last: BlockNumberFor::<Test>::from(15u32),
+            })
+        );
+    });
+}
+
+#[test]
+fn assign_core_with_end_hint() {
```

Comment:
This doesn't actually test the behaviour, just that the schedule makes it in. I'd suggest running to the start block, making sure it's in the CoreDescriptors and then running to the last block to make sure that it's removed from `current_work`